### PR TITLE
update Ignore docs

### PR DESCRIPTION
**Pull request by khorne3**
For [TEC-5](https://linear.app/semgrep/issue/TEC-5/interactions-of-various-semgrepignores-should-scp-add-to-override):

* Fixed confusing section on how to ignore files during triage
* Updated summary chart to reflect current behavior, specifically regarding gitignore
* Clarified how the three ignore lists interact -- this is complicated and may need additional edits. Please let me know if you find my new explanations clear as mud.

For [TEC-53](https://linear.app/semgrep/issue/TEC-53/update-ignore-docs-with-information-on-pre-product-ignores):

* Updated instructions on how to use the Path Ignores feature in the UI on the Ignores page and the Secrets page

For [MKT-605](https://linear.app/semgrep/issue/MKT-605/troubleshooting-ignores-semgrepignore-vs-ignore-in-app):

* Added explicit instructions on creating `.semgrepignore` and updating Path Ignores box to include files that would otherwise be ignored by Semgrep

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR

### CODE REVIEW

**Review by salolivares — COMMENTED** [review]
Docs look good so far :) will approve once out of draft!

File: docs/ignore-oss.md
Comment (khorne3) [inline docs/ignore-oss.md:66]: I just realized this isn't a part of the sidebars; we have a redirect in place, so should we just delete this doc?

Comment (armchairlinguist) [reply docs/ignore-oss.md:66]: where is it redirecting to? I think this doc is quite useful, but if we have similar content elsewhere...?

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:102]:
```suggestion
4. To define files and folders that Semgrep Code and Semgrep Supply Chain ignore:
```

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:113]:
```suggestion
Adding items to Semgrep AppSec Platform's **Path Ignores** box doesn't override default Semgrep ignore patterns included with its CLI tool, since the patterns are additive. However, items added to `.semgrepignore` override default Semgrep CLI patterns.
```

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:209]: This whole section doesn't make sense to me as it's written - are users trying to exclude or include?

I think based on what the variable is called, all of this should be written to exclude the paths.

File: docs/ignore-oss.md
Comment (armchairlinguist) [reply docs/ignore-oss.md:66]: gotcha - in that doc, do we make clear which options are oss-available? if so then yeah I don't see the point of keeping this

File: docs/ignoring-files-folders-code.md
Comment (khorne3) [reply docs/ignoring-files-folders-code.md:209]: I might have gotten tunnel vision here: the person asking the question wanted to include their test folder, which Semgrep excludes by default. They tried to work around this by explicitly stating what to include (the test folder) using the env var, but it didn't work because Semgrep ignored the env var.

To simplify things, I will focus on excluding paths (and how to make it so that things aren't excluded) instead of how to include things.

File: docs/ignore-oss.md
Comment (khorne3) [reply docs/ignore-oss.md:66]: The options aren't explicitly flagged as OSS, but if they require SAP, we make that clear in the headers

Comment (armchairlinguist) [reply docs/ignore-oss.md:66]: sounds perfect!

**Review by s-santillan — APPROVED** [review]

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:67]: I think this document covers `semgrep ci` scans as well; if that's the case, we should probably mention the project ignores in this item as well, as they will still be sent down and many of them overlap with the default ignores from the CLI (which is what .semgrepignore overrides).

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:71]: Is it worth mentioning that `--include` will exclude everything that doesn't match, as well?

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:107]:
```suggestion
Including files and folders through this method is **additive**. When a `semgrep ci` scan is run, it looks for a `.semgrepignore` within the repository. If no `.semgrepignore` file is found, Semgrep temporarily creates one and adds items from Semgrep AppSec Platform's Path Ignores.
```
I'm not particular about the wording, but the platform doesn't initiate scans, the CLI does. (Managed scanning is a partial exception to this, but things like ignores which are part of the config are still a response to a request from the CLI to start a scan.)

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:113]: This doesn't appear to be possible from the Findings list view anymore. It is possible from either grouping mode once you get to the finding details page (since that's just a page for one finding either way).

I'm not actually sure if this is intended behavior, so it could be worth following up with the team. But for now I would say just change this to "while triaging individual findings using Semgrep AppSec Platform" and then say that you first go to the Details page in Step 2 below.

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:207]:
```suggestion
1. Create a `.semgrepignore` file that lists the files you want excluded.
```

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:208]: I'm pretty sure one or the other of these is sufficient (doing both is overkill, since they're additive), so we should probably be explicit about that.

File: docs/ignoring-files-folders-code.md
Comment (armchairlinguist) [inline docs/ignoring-files-folders-code.md:67]:
```suggestion
| To scan all files within Semgrep's scope each time you run Semgrep (only files within `.git` are ignored). | Create an empty `.semgrepignore` file in your repository root directory or in your project's working directory, and for `semgrep ci` scans, [remove any entries listed in your project's **Path Ignores** list](#define-ignored-files-and-folders-in-semgrep-appsec-platform)  in Semgrep AppSec Platform. |
```
again not picky about wording here, but for platform-enabled scans both are required

**Review by armchairlinguist — COMMENTED** [review]
just one more small thing :)

**Comment by khorne3** [comment]
@armchairlinguist thank you for all of your help!

**Review by salolivares — APPROVED** [review]
LGTM!! 

I wouldn't merge just yet, i'm planning to rollout the feature tomorrow! Once rolled out to everyone we can merge the docs :)

**Comment by salolivares** [comment]
@khorne3 this has been rolled out to the main tenant! still working on the customer specific tenants. 

Can we merge?

**Comment by khorne3** [comment]
@salolivares Yup, will do so now

**Pull request merged**
